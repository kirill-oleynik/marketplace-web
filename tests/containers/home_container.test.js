import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import HomeContainer from 'app/containers/home_container';

const fakeStore = (state = {}) => ({
  subscribe() {},
  dispatch() {},
  getState() { return state; }
});

const application = {
  id: 1,
  title: 'Test1'
};

const category = {
  id: 2,
  title: 'Test2'
};

const applications = {
  ids: [1],
  byId: {
    1: application
  }
};

const categories = {
  byId: {
    2: category
  }
};

const search = {
  categoryIds: [2],
  applicationIds: [1]
};

describe('#render', () => {
  test('it renders correctly', () => {
    const store = fakeStore({
      search,
      categories,
      applications
    });

    const component = shallow(
      <HomeContainer />,
      { context: { store } }
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#passedProps', () => {
  test('it passess props to component correctly', () => {
    const store = fakeStore({
      search,
      categories,
      applications
    });

    const component = shallow(
      <HomeContainer />,
      { context: { store } }
    );

    expect(
      component.props().featuredApplications
    ).toEqual([application]);

    expect(
      component.props().searchCategories
    ).toEqual([category]);

    expect(
      component.props().searchApplications
    ).toEqual([application]);
  });
});
