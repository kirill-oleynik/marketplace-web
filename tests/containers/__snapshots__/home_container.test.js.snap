// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<Translate(WithFixedHeader)
  featuredApplications={
    Array [
      Object {
        "id": 1,
        "title": "Test1",
      },
    ]
  }
  fetch={[Function]}
  searchApplications={
    Array [
      Object {
        "id": 1,
        "title": "Test1",
      },
    ]
  }
  searchCategories={
    Array [
      Object {
        "id": 2,
        "title": "Test2",
      },
    ]
  }
/>
`;
